### DISCUSSION TITLE
What is the best way to add a large static game level into the physics system?

### DISCUSSION THREAD

**Discussion by Mahmoud1205** [jrouwe/JoltPhysics#1298] [Q&A] [answered]
The game level is completely static, nothing moves inside it until now.
I have three ideas, first is to merge all the level's meshes into one then add that as a mesh shape into the physics system, second is to make a static compound shape consisting of the level's meshes as mesh shapes, third is same as the second but each separate entity is a static compound shape, meaning each building or tree is an independent body.

**Comment by jrouwe** [comment, marked as answer]
> First is to merge all the level's meshes into one then add that as a mesh shape into the physics system

There's a limit to the number of triangles that fit into a single mesh, so you will eventually run into problems with this.

> Second is to make a static compound shape consisting of the level's meshes as mesh shapes

Again, this could run into issues as the 32-bit SubShapeID may run out of space.

> Third is same as the second but each separate entity is a static compound shape, meaning each building or tree is an independent body

That could work. Alternatively, you could cluster entities and create a single body per cluster. E.g. you could pick areas of 50x50 m and add everything in that area in a single compound shape/body. Use e.g. the center point of the entity to classify (so the entities could stick out of the 50x50 area a bit).

**Reply by Mahmoud1205** [reply]
Okay got it. Thanks.
